### ISSUE TITLE
Change keyboard shortcuts?

### ISSUE THREAD

**Issue by NoLooseEnds** [silverbulletmd/silverbullet#193]
Is there any way to change the keyboard shortcuts by changing the config? I'm on a Norwegian keyboard, and our `/`-key is "hidden" behind `shift + 7`. Using `cmd + shift + 7` does not work, so I'm forced using the mouse to click which influences the workflow in a negative way.

![image](https://user-images.githubusercontent.com/4103531/207343128-1cf9dfbb-d14f-45da-aded-b541722dd938.png)

Bonus question:
Is there possible to change the default font in the config, without overriding the CSS?

**Comment by zefhemel** [comment]
The answer to both of these for now is: no, not yet possible.

Regarding keyboard mappings: yeah, this is something I know we need to support at some point. I’m open to how to configure this though. Perhaps it would make sense to put this in YAML format in `SETTINGS` something along the lines of:

```yaml
keys:
  “Command Palette”:
    mac: “Cmd-p”
    key: “Ctrl-p”
  “Page: Rename”:
    mac: “Cmd-r”
    # etc.
```

Command patte would have to be a special case because it’s not a command itself. Another approach would be to mirror the struction of plug files but then allow specific overrides. This would be more flexible (you could override various other things), but probably harder to use, e.g.:

```yaml
overrides:
   core:
      renamePage:
        command:
            mac: “Cmd-r”
```

Basically this way, we’d merge whatever you keep under `overrides` and then the plug name (e.g. `core`) with what it pulls from e.g. https://github.com/silverbulletmd/silverbullet/blob/main/plugs/core/core.plug.yaml allowing you to override arbitrary attributes. This way you could also rename commands for instance, or even define basic new ones.

Regarding overriding fonts: also not possible right now. There’s a discussion on theming here, it just needs to be picked up: https://github.com/silverbulletmd/silverbullet/discussions/58 For now a lot of the lay-out working relies on monspaced fonts, so that will likely remain a restriction for some time.

**Comment by NoLooseEnds** [comment]
As long as the documentation is on par, both should work. Especially if you map out the most common use cases, and explain how it's done.

**Comment by ryanwwest** [comment]
Settings seems like a good place to put this, also makes it easier to migrate your settings+keybindings configs to another computer when necessary. I like the [sioyek](https://github.com/ahrm/sioyek) academic/vim PDF reader's approach to this, which has a [default keybindings config file](https://github.com/ahrm/sioyek/blob/main/pdf_viewer/keys.config) with tons of comments for EVERY keybinding and a [user keybindings config file](https://github.com/ahrm/sioyek/blob/main/pdf_viewer/keys_user.config) that allows the user to override **any** of the defaults and add multiple keybindings per action (and they have the same default+user file pair option for all remaining settings). Super clear (basic documentation is inline and often suffices) and customizable.

**Comment by ryanwwest** [comment]
Of course it most likely wouldn't include any vim keybindings since those are handled out of this project. In an ideal world I could see some sort of warning that pops up when you assign an action to a keybinding which is already used by 1) an existing Silverbullet action or 2) an existing vim action. Perhaps being able to provide this feedback is one reason why Obsidian has a dedicated UI for hotkeys and not a plain file... But I still think the file-based approach could work here.

**Comment by zefhemel** [comment]
Alright, this is now implemented. However, it only allows you to override keyboard shortcuts for commands, not for the built-in two shortcuts for the command palette and page picker, everything else can be overridden, more than that, this even enables you to define custom commands and slash commands on the fly! I need to document that better, but there's some hints at the example SETTINGS here: https://silverbullet.md/SETTINGS
